core/src/lightclient/mod.rs

```diff
@@ -354,4 +367,130 @@ impl StarknetClient for StarknetClientImpl {
+    ///
+    /// `Ok(Option<TransactionReceipt>)` if the operation was successful.
+    /// `Err(LightClientError)` if the operation failed.
+    async fn get_transaction_receipt(
+        &self,
+        hash: H256,
+    ) -> Result<Option<TransactionReceipt>, LightClientError> {
+        let mut res_receipt = MyTransactionReceipt::default().0;
+
+        //TODO: Error when trying to transform 32 bytes hash to FieldElement
+        let hash_hex = hex::encode(hash);
+        let hash_felt = FieldElement::from_hex_be(&hash_hex).map_err(|e| {
+            LightClientError::OtherError(anyhow::anyhow!(
+                "Failed to convert Starknet block hash to FieldElement: {}",
+                e
+            ))
+        })?;
+        let starknet_tx_receipt = self.client.get_transaction_receipt(hash_felt).await?;
+
+        match starknet_tx_receipt {
+            MaybePendingTransactionReceipt::Receipt(receipt) => {
+                match receipt {
+                    StarknetTransactionReceipt::Invoke(InvokeTransactionReceipt {
+                        transaction_hash,
+                        actual_fee,
+                        status,
+                        block_hash,
+                        block_number,
+                        ..
+                    })
+                    | StarknetTransactionReceipt::L1Handler(L1HandlerTransactionReceipt {
+                        transaction_hash,
+                        actual_fee,
+                        status,
+                        block_hash,
+                        block_number,
+                        ..
+                    })
+                    | StarknetTransactionReceipt::Declare(DeclareTransactionReceipt {
+                        transaction_hash,
+                        actual_fee,
+                        status,
+                        block_hash,
+                        block_number,
+                        ..
+                    })
+                    | StarknetTransactionReceipt::Deploy(DeployTransactionReceipt {
+                        transaction_hash,
+                        actual_fee,
+                        status,
+                        block_hash,
+                        block_number,
+                        ..
+                    })
+                    | StarknetTransactionReceipt::DeployAccount(
+                        DeployAccountTransactionReceipt {
+                            transaction_hash,
+                            actual_fee,
+                            status,
+                            block_hash,
+                            block_number,
+                            ..
+                        },
+                    ) => {
+                        res_receipt.transaction_hash =
+                            Some(PrimitiveH256::from_slice(&transaction_hash.to_bytes_be()));
+                        res_receipt.gas_used = Some(felt_to_u256(actual_fee));
```

Comment:
`actual_fee` is not exactly `gas_used` 
https://docs.starknet.io/documentation/architecture_and_concepts/Fees/fee-mechanism/

I think actual_fee is gas_used * fee